### PR TITLE
Compatibility with Julia 0.6

### PR DESCRIPTION
**Pull request by omus**

### CODE REVIEW

**Comment by omus** [comment]
Fixes #44

File: REQUIRE [hunk 1]
```diff
@@ -1,5 +1,5 @@
 julia 0.4
-Compat 0.8
+Compat 0.9.4
```
Comment (tkelman) [inline REQUIRE]: 0.9.5 for `take!`

Comment (omus) [inline REQUIRE]: Thanks

**Comment by omus** [comment]
I have one more commit to push which utilizes the changes in https://github.com/JuliaLang/julia/pull/20952. 

The commit "Temporarily restrict parsing pre-Julia 0.6" should work on 0.6 as a version of TimeZones.jl without parsing support. Currently it's failing because `Dates.parse(::AbstractString, ::DateFormat)` is missing which is required for compiling the tzdata.

**Comment by omus** [comment]
Testing commit which utilizes changes in https://github.com/JuliaLang/julia/pull/20952.

I'm not sure if the latest nightly includes the changes yet.

**Pull request merged**